com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/HDProjectSettings.cs

```diff
@@ -0,0 +1,338 @@
+        /// <by cref="IPluginSubTargetMaterialUtils"/>.
+        /// <See also cref="MaterialReimporter.RegisterUpgraderReimport"/>
+        /// </summary>
+        [NonSerialized]
+        long m_InProjectPluginLastSeenMaterialVersionsSum = (int)PluginMaterial.GenericVersions.NeverMigrated;
+
+        [NonSerialized]
+        long m_InProjectPluginLastSeenSubTargetVersionsSum = (int)PluginMaterial.GenericVersions.NeverMigrated;
+
+        internal const int k_NeverProcessedMaterialVersion = -1;
+
+        public static string projectSettingsFolderPath
+        {
+            get => instance.m_ProjectSettingFolderPath;
+            set
+            {
+                instance.m_ProjectSettingFolderPath = value;
+                Save();
+            }
+        }
+
+        public static bool wizardIsStartPopup
+        {
+            get => instance.m_WizardPopupAtStart;
+            set
+            {
+                instance.m_WizardPopupAtStart = value;
+                Save();
+            }
+        }
+
+
+        public static int materialVersionForUpgrade
+        {
+            get => instance.m_LastMaterialVersion;
+            set
+            {
+                instance.m_LastMaterialVersion = value;
+                Save();
+            }
+        }
+
+        public static ShaderGraphVersion hdShaderGraphLastSeenVersion
+        {
+            get => instance.m_HDShaderGraphLastSeenVersion;
+            set
+            {
+                instance.m_HDShaderGraphLastSeenVersion = value;
+                Save();
+            }
+        }
+
+        public static long pluginSubTargetLastSeenMaterialVersionsSum
+        {
+            get => instance.m_InProjectPluginLastSeenMaterialVersionsSum;
+        }
+
+        public static long pluginSubTargetLastSeenSubTargetVersionsSum
+        {
+            get => instance.m_InProjectPluginLastSeenSubTargetVersionsSum;
+        }
+
+        //singleton pattern
+        static HDProjectSettings s_Instance;
+        static HDProjectSettings instance => s_Instance ?? CreateOrLoad();
+
+        HDProjectSettings()
+        {
+            s_Instance = this;
+            // s_Instance.FillPresentPluginMaterialVersions(); <
+            // We can't call this here as the scriptable object will not be deserialized in yet
+        }
+
+        // We force the instance to be loaded/created and ready with valid values on assembly reload.
+        // We also use this so that the HDUserSettings.cs on the runtime assembly will have the HDProjectSettings proxy
+        // injected before it is used.
+        [InitializeOnLoadMethod]
+        static void Reset()
+        {
+            // Make sure the cached last seen plugin versions (capped to codebase versions) and their sum is valid
+            // on assembly reload.
+            instance.FillPresentPluginMaterialVersions();
+            HDProjectSettingsProxy.projectSettingsFolderPath = () => projectSettingsFolderPath;
```

Comment:
@sebastienlagarde 
InitializeOnLoadMethod with this dependency injection is what allowed me to move back HDProjectSettings.cs into editor land and solve the snafu created between this PR and this one https://github.com/Unity-Technologies/Graphics/pull/4389 

@RSlysz for visibility.